src/v/storage/kvstore.h

```diff
@@ -111,6 +111,16 @@ class kvstore {
     ss::future<> start();
     ss::future<> stop();
 
+    /*
+     * Perform deferred GC operations, saving a kvstore snapshot and removing
+     * replayed segments, which are no longer needed.
+     *
+     * This is deferred from replay_segments to avoid modifying local data
```

Comment:
mega nit: This is deferred from replaying segments **on startup** to avoid...

Reply:
Indeed. Done.